controllers/controllervm_controller.go

```diff
@@ -156,18 +177,30 @@ func (r *ControllerVMReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		// TODO: multi nic support with bindata template
 		// TODO: for now single network and hardcode controller ips to 192.168.25.1X
 
-		hostname := fmt.Sprintf("controller-%d", i)
+		hostKey := fmt.Sprintf("%s%d", instance.Name, i)
+		hostname, err := common.CreateOrGetHostname(instance, hostKey, instance.Name)
+		r.Log.Info(fmt.Sprintf("Controlplane VM hostname set to %s", hostname))
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		//hostname := fmt.Sprintf("controller-%d", i)
 		networkDataSecretName := fmt.Sprintf("%s-%s-networkdata", instance.Name, hostname)
 
+		ip, _, _ := net.ParseCIDR(ipset.Status.HostIPs[hostKey].IPAddresses["ctlplane"]) // We use ctlplane as the MgmtNetwork too for now
+		// FIXME where is the netmask set!?
```

Comment:
the controllers use cloud-init networking config v2 https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v2.html#common-properties-for-all-device-types , so it uses CIDR notation https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/templates/controllervm/cloudinit/networkdata#L4

Reply:
I think I got it in the latest patch. It is an IP/netmask so I don't even need to use net.ParseCIDR format on it. Its ready to go straight from the ipset...